wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java

```diff
@@ -378,10 +379,15 @@ static public boolean isWorkunitComplete(WUState state)
      */
     public static WUState getStateID(WorkunitInfo wu)
     {
-        if (wu != null)
-            return getStateID(wu.getStateID());
-        else
-            return WUState.UNKNOWN;
+        if (wu != null) {
+            if (wu.getStateID() != null) {
```

Comment:
@lrenn please either remove the curly brackets around the if and else if blocks, or move the open curly to a new line as well as the else if, and squash your commits into 1 so we can finalize this pull request. Thanks.